bin/copyUserPostVoteData.ts

```diff
@@ -0,0 +1,92 @@
+import '../src/config';
+import createOrGetConnection from '../src/db';
+import { Downvote, Upvote } from '../src/entity';
+
+(async (): Promise<void> => {
+  const entityArgument = process.argv[2];
+  const createdFromArgument = process.argv[3];
+  const createdToArgument = process.argv[4];
+
+  const argumentToEntityMap = {
+    Upvote: Upvote,
+    Downvote: Downvote,
+  };
+  const entity = argumentToEntityMap[entityArgument];
+
+  if (!entity) {
+    throw new Error(
+      `Invalid entity, select one from: ${Object.keys(argumentToEntityMap).join(
+        ', ',
+      )}`,
+    );
+  }
+
+  if (!createdFromArgument || !createdToArgument) {
+    throw new Error('createdFrom and createdTo arguments are required');
+  }
+
+  const createdFromDate = new Date(createdFromArgument);
+
+  if (Number.isNaN(createdFromDate.getTime())) {
+    throw new Error(
+      'createdFromDate argument is invalid, format should be ISO 6801',
+    );
+  }
+
+  const createdToDate = new Date(createdToArgument);
+
+  if (Number.isNaN(createdToDate.getTime())) {
+    throw new Error(
+      'createdToDate argument is invalid, format should be ISO 6801',
+    );
+  }
+
+  if (createdFromDate > createdToDate) {
+    throw new Error(
+      'createdFrom argument should be less than createdTo argument',
+    );
+  }
+
+  const con = await createOrGetConnection();
+
+  await con.transaction(async (manager) => {
+    switch (entity) {
+      case Upvote:
+        await manager.query(`
+          INSERT INTO user_post ("postId", "userId", vote)
+          SELECT "postId", "userId", 1 AS vote
+          FROM upvote
+          WHERE "createdAt" > '${createdFromDate.toISOString()}' AND "createdAt" < '${createdToDate.toISOString()}'
+          ON CONFLICT ("postId", "userId") DO UPDATE SET vote = 1;
+        `);
+        await manager.query(`
+          INSERT INTO user_post ("postId", "userId", "votedAt")
+          SELECT "postId", "userId", "createdAt" as "votedAt"
+          FROM upvote
+          WHERE "createdAt" > '${createdFromDate.toISOString()}' AND "createdAt" < '${createdToDate.toISOString()}'
+          ON CONFLICT ("postId", "userId") DO UPDATE SET "votedAt" = EXCLUDED."votedAt";
+        `);
```

Comment:
Additional query here is to set `votedAt` per old table `createdAt`. We can not set it in one query because `voted_at_time` function in triggers will always overwrite to `now()`. I think it should be fine as long as we don't do many updates in a chunk. For info, we did 100k per chunk easily in posts flags migration.

Reply:
We can also temporary disable triggers for this script and re-enable at the end?
https://stackoverflow.com/questions/3942258/how-do-i-temporarily-disable-triggers-in-postgresql#:~:text=Alternatively%2C%20if%20you%20are%20wanting,triggers%20for%20the%20current%20session.

Reply:
Since we don't have a lot of data I believe we don't even need to split it to different batches. Happy with the current implementation even if it requires an extra query. It's just a one off